### PR TITLE
chore: typescript setup

### PR DESCRIPTION
**Pull request by Donya-qlik**

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-  testEnvironment: "jsdom",
+  testEnvironment: "jest-environment-jsdom",
```
Comment (Donya-qlik) [inline jest.config.js:2]: It's required for jest> 28

File: package.json [hunk 2]
```diff
@@ -73,6 +73,12 @@
       ]
     }
   },
+  "dependencies": {
+    "@types/jest": "^29.5.5",
+    "@types/node": "^20.7.1",
+    "jest-environment-jsdom": "^29.7.0",
+    "typescript": "^5.2.2"
```
Comment (niekvanstaveren) [inline package.json:80]: These should be dev dependencies and locked

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "compilerOptions": {
+    "jsx": "react",
```
Comment (niekvanstaveren) [inline tsconfig.json:3]: There is no react

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (niekvanstaveren) [inline tsconfig.json:1]: Can we extend qlik tsconfig? It is added as dependency...

File: package.json [hunk 5]
```diff
@@ -86,17 +92,17 @@
     "@nebula.js/cli-serve": "4.4.0",
     "@nebula.js/stardust": "4.4.0",
     "@nebula.js/test-utils": "0.6.0",
-    "@playwright/test": "1.38.1",
-    "@qlik/eslint-config": "0.4.21",
-    "@qlik/prettier-config": "0.4.0",
+    "@playwright/test": "1.38.0",
```
Comment (niekvanstaveren) [inline package.json:95]: This is a downgrade...

File: package.json [hunk 6]
```diff
@@ -86,17 +92,17 @@
...
+    "@qlik/eslint-config": "0.4.18",
+    "@qlik/prettier-config": "0.3.0",
```
Comment (niekvanstaveren) [inline package.json:97]: This is a downgrade. A specific reason here?

**Review by niekvanstaveren — CHANGES_REQUESTED** [review]

**Comment by niekvanstaveren** [comment]
To fully test this you will need to convert a file to typescript as well. Also a script to check the types would be good to have in that case

File: package.json (hunk 2 above)
Comment (Donya-qlik) [reply package.json]: Moved them to devDep

**Comment by Donya-qlik** [comment]
> To fully test this you will need to convert a file to typescript as well. Also a script to check the types would be good to have in that case

Should it be a specific file for this? 
what kind of script do you mean ? unit test ?

**Comment by niekvanstaveren** [comment]
> > To fully test this you will need to convert a file to typescript as well. Also a script to check the types would be good to have in that case
> 
> Should it be a specific file for this? what kind of script do you mean ? unit test ?

No it can probably be any file. Look for something that could really benefit from being typed I guess.
I am talking about a script in package.json. If you look at other repos they will have an entry:

`types:check: "tsc --noEmit"` (Or something similar)

**Comment by Donya-qlik** [comment]
I decided to close this PR temporarily until the refactoring [PR](https://github.com/qlik-oss/sn-action-button/pull/436) is merged

**Pull request closed without merging**